tests/media/test_media_storage.py

```diff
@@ -257,10 +258,15 @@ def write_to(
                 output_stream.write(data)
                 return response
 
+            def write_err(f: Failure) -> Failure:
+                f.trap(HttpResponseException)
+                output_stream.write(f.value.response)
+                return f
```

Comment:
This one was new to me: https://docs.twisted.org/en/stable/api/twisted.python.failure.Failure.html#trap

TL;DR the trap call is a no-op if `f ` contains an `HTTPResponseException`; otherwise the trap raises immediately so that the next errback can handle this Failure.

Reply:
I'd missed that this was in test code though. Why do we need to add this as an errback all of a sudden?

Reply:
We need to add it because we know call `errback` sometimes on the list of Deferreds so that we can resolve a request with an error instead of a response.

Reply:
See https://github.com/matrix-org/synapse/pull/16701/files#diff-50994e96eaadcf21dd111024e283f06c231adc413a3530ce7356e02ada339a3eR704-R706

Reply:
Ahhh, I think I see: we never called errback on the mock until now!